Fluid/Ast/ForStatement.cs

```diff
@@ -59,7 +63,18 @@ public override async Task<Completion> WriteToAsync(TextWriter writer, TextEncod
             {
                 int start = Convert.ToInt32((await Range.From.EvaluateAsync(context)).ToNumberValue());
                 int end = Convert.ToInt32((await Range.To.EvaluateAsync(context)).ToNumberValue());
-                elements = Enumerable.Range(start, end - start + 1).Select(x => new NumberValue(x));
+
+                // Cache range
+                if (_rangeElements == null || _rangeStart != start || _rangeEnd != end)
+                {
+                    _rangeElements = elements = Enumerable.Range(start, end - start + 1).Select(x => new NumberValue(x));
```

Comment:
Do you want to cache the initial projection template which really does nothing or the one that `ToList` produces at the end?

Reply:
Ah I see that Limit etc can be dynamic, this can be hard to cache then. Obvious logic to cache the list would be if all logic rules are literal or null? Don't know if that's the usual case.